torch/csrc/jit/codegen/cuda/test/test_gpu_transpose.cpp

```diff
@@ -0,0 +1,466 @@
+  fusion.addInput(tv1);
+  auto tv2 = transpose(tv0, 0, 2);
+  auto tv3 = sin(tv2);
+  fusion.addOutput(tv3);
+  auto tv4 = add(tv0, tv1);
+  auto tv5 = cos(tv4);
+  fusion.addOutput(tv5);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input0 = at::randn({256, 1024, 1024}, options);
+  at::Tensor input1 = at::randn({256, 1024, 1024}, options);
+
+  auto lparams = scheduleTranspose(&fusion, {input0, input1});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input0, input1}, lparams);
+  auto outputs = fe.runFusion({input0, input1}, lparams);
+
+  auto tv_ref1 = input0.transpose(0, 2).sin();
+  auto tv_ref2 = (input0 + input1).cos();
+
+  testValidate(
+      &fusion,
+      outputs,
+      {input0, input1},
+      {tv_ref1, tv_ref2},
+      __LINE__,
+      __FILE__);
+}
+
+//             .------>sin------>z
+// x->transpose->transpose->add->y
+//  \_______________________/
+TEST_F(NVFuserTest, FusionScheduleTransposeMatchingSkipConnection_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(3);
+  fusion.addInput(tv0);
+  auto tv1 = transpose(tv0, 0, 2);
+  auto tv2 = transpose(tv1, 0, 2);
+  auto tv3 = add(tv0, tv2);
+  fusion.addOutput(tv3);
+  auto tv4 = sin(tv1);
+  fusion.addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input = at::randn({256, 1024, 1024}, options);
+
+  auto lparams = scheduleTranspose(&fusion, {input});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input}, lparams);
+  auto outputs = fe.runFusion({input}, lparams);
+
+  auto tv_ref1 = input.transpose(0, 2).transpose(0, 2) + input;
+  auto tv_ref2 = input.transpose(0, 2).sin();
+
+  testValidate(
+      &fusion, outputs, {input}, {tv_ref1, tv_ref2}, __LINE__, __FILE__);
+}
+
+// x->transpose--add->z
+// y->broadcast-/
+TEST_F(NVFuserTest, FusionScheduleTransposeBroadcast_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(3);
+  auto tv1 = makeContigTensor(2);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+  auto tv2 = transpose(tv0, 1, 2);
+  auto tv3 = broadcast(tv1, {false, false, true});
+  auto tv4 = add(tv2, tv3);
+  fusion.addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input0 = at::randn({1024, 256, 1024}, options);
+  at::Tensor input1 = at::randn({1024, 1024}, options);
+
+  auto lparams = scheduleTranspose(&fusion, {input0, input1});
+  // auto lparams = schedulePointwise(&fusion, {input0, input1});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input0, input1}, lparams);
+  auto outputs = fe.runFusion({input0, input1}, lparams);
+
+  auto tv_ref = input0.transpose(1, 2) + input1.unsqueeze(2);
+
+  testValidate(
+      &fusion, outputs, {input0, input1}, {tv_ref}, __LINE__, __FILE__);
+}
+
+// x->broadcast--add->z
+// y->broadcast-/
+TEST_F(NVFuserTest, FusionScheduleBroadcastOnly_CUDA) {
```

Comment:
Indexing error, working on a minimum repro

Reply:
#1873 to track this one.

Reply:
Thanks, @shmsong, for the help and fast response on this. I think both the scheduler and the indexing are doing something wrong. I will push new commits to my scheduler to detect this and error out early, and #1873 should be sufficient to track the indexing issue.